### PR TITLE
[Backport] Upgrade reactor netty http to v1.0.39 and reactor core to v3.4.34

### PR DESCRIPTION
**Pull request by slobodanadamovic**
This PR upgrades the version of `reactor-netty-http` library to the latest v1.0.39 version
and its transitive dependencies `reactor-core` to v3.4.34 
and `reactor-netty-core` to v1.0.39.

Backport of https://github.com/elastic/elasticsearch/pull/102311

### CODE REVIEW

**Comment by slobodanadamovic** [comment]
Note: Marking this PR as `draft` until I figure out where these old dependencies are coming from: https://github.com/elastic/elasticsearch/pull/102629/commits/a0f1ccdb1da98ac7b5842c0bb32f12dbef704f80

**Comment by slobodanadamovic** [comment]
> Note: Marking this PR as `draft` until I figure out where these old dependencies are coming from: [a0f1ccd](https://github.com/elastic/elasticsearch/commit/a0f1ccdb1da98ac7b5842c0bb32f12dbef704f80)

Turns out they are transient dependencies introduced by `build-tools-internal`. Marking this PR as ready since they do not affect production code.
image

File: docs/changelog/102629.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 102629
+summary: "[Backport] Upgrade reactor netty http version to 1.0.39"
```
Comment (kingherc) [inline docs/changelog/102629.yaml:2]: nit and version 3.4.34 inside these

**Review by kingherc — APPROVED** [review]
LGTM

**Comment by slobodanadamovic** [comment]
@elasticmachine update branch

**Pull request merged**